test/arbitrable-permission-list.js

```diff
@@ -0,0 +1,634 @@
+    );
+    await expectThrow(
+      arbitrablePL.requestRegistration(ITEM_VALUE, {
+        from: submitter,
+        value: stake + arbitrationFee - 1
+      })
+    );
+    await expectThrow(
+      arbitrablePL.requestClearing(ITEM_VALUE, {
+        from: submitter,
+        value: stake + arbitrationFee - 1
+      })
+    );
+    await expectThrow(
+      arbitrablePL.challengeRegistration(ITEM_VALUE, {
+        from: challenger,
+        value: stake + arbitrationFee - 1
+      })
+    );
+    await expectThrow(
+      arbitrablePL.challengeClearing(ITEM_VALUE, {
+        from: challenger,
+        value: stake + arbitrationFee - 1
+      })
+    );
+  });
+
+  it("Should create a dispute when someone challenges registration", async () => {
+    const centralizedArbitrator = await CentralizedArbitrator.new(
+      arbitrationFee,
+      { from: arbitrator }
+    );
+    const arbitrablePL = await ArbitrablePermissionList.new(
+      centralizedArbitrator.address,
+      arbitratorExtraData,
+      metaEvidence,
+      blacklist,
+      appendOnly,
+      rechallengePossible,
+      stake,
+      timeToChallenge,
+      { from: submitter }
+    );
+
+    await arbitrablePL.requestRegistration(ITEM_VALUE, {
+      from: submitter,
+      value: stake + arbitrationFee
+    });
+    //await arbitrablePL.executeRequest(ITEM_VALUE,{from:submitter});
+    await arbitrablePL.challengeRegistration(ITEM_VALUE, {
+      from: challenger,
+      value: stake + arbitrationFee
+    });
+    const dispute = await centralizedArbitrator.disputes(0);
+    assert.equal(
+      dispute[0],
+      arbitrablePL.address,
+      "Arbitrable not set up properly"
+    );
+    assert.equal(
+      dispute[1].toNumber(),
+      2,
+      "Number of choices not set up properly"
+    );
+    assert.equal(
+      dispute[2].toNumber(),
+      arbitrationFee,
+      "Fee not set up properly"
+    );
+  });
+  // it('Should not be able to challenge registration when arbitration fee increases', async () => {
+  //   const centralizedArbitrator = await CentralizedArbitrator.new(
+  //     arbitrationFee,
+  //     { from: arbitrator }
+  //   );
+  //   const arbitrablePL = await ArbitrablePermissionList.new(
+  //     centralizedArbitrator.address,
+  //     arbitratorExtraData,
+  //     metaEvidence,
+  //     blacklist,
+  //     appendOnly,
+  //     rechallengePossible,
+  //     stake,
+  //     timeToChallenge,
+  //     { from: submitter }
+  //   );
+
+  //   const submitterTx = await arbitrablePL.requestRegistration(ITEM_VALUE, {
+  //     from: submitter,
+  //     value: stake + arbitrationFee,
+  //     gasPrice: gasPrice
+  //   });
+  //   const submitterTxFee = submitterTx.receipt.gasUsed * gasPrice;
+  //   const submitterBalBeforeReqCancel = web3.eth.getBalance(submitter);
+  //   const challengerBalBeforeReqCancel = web3.eth.getBalance(challenger);
+  //   const challengerTx = await arbitrablePL.challengeRegistration(ITEM_VALUE, {
```

Comment:
We are missing an increase in fees before this call, that is why `item.balance >= arbitratorCost` in the contract is returning true and the test is failing. We can see this by adding: 
```
    console.info(`arbitrationCost`, (await centralizedArbitrator.arbitrationCost(``)).toNumber())
    console.info(`itemBalance`, (await arbitrablePL.items(ITEM_VALUE))[4].toNumber())
```
To fix this issue and continue working, call
```
    await centralizedArbitrator.setArbitrationPrice(16, { from: arbitrator })
```
before calling `challengeRegistration()`.
The `16` parameter could be any value higher than itemBalance.

Reply:
still its failing. I tried itemBalance is 15

Reply:
Now the problem is something else:

Seems to be that `require(msg.value >= stake + arbitratorCost, "Not enough ETH.");` is causing a revert because the value of `arbitrationFee` is outdated here:

https://github.com/kleros/kleros-interaction/blob/b9058277cf2822d0cdd6bd4da0b864c09121f05e/test/arbitrable-permission-list.js#L221

Fixing that should make the transaction go through.

Reply:
I didnt understand... how the arbitration price is outdated?

Reply:
The value of the arbitrationFee is 5 (Line 22), which is set to the arbitrator contract whenever we instantiate it. However, we increase the cost to 17 at line 218 for our test. So when call:
```
await arbitrablePL.challengeRegistration(ITEM_VALUE, {
      from: challenger,
      value: stake + arbitrationFee,
      gasPrice: gasPrice
});
```
we are sending less then what is necessary:
msg.value is 10 (stake) + 5 (arbitrationFee), when we should really be sending 10 + 17.

When the contract execution reaches line 182: 
```
item.disputeID = arbitrator.createDispute.value(arbitratorCost)(2,arbitratorExtraData);
```
it reverts because `requireArbitrationFee` in the Arbitrator.sol contract fails.
One solution is to use this instead:
```
await arbitrablePL.challengeRegistration(ITEM_VALUE, {
      from: challenger,
      value: stake + 17,
      gasPrice: gasPrice
});
```